traitsui/qt4/range_editor.py

```diff
@@ -208,8 +215,12 @@ def update_object_on_enter(self):
                 )
             finally:
                 self.control.slider.blockSignals(blocked)
-        except TraitError as excp:
-            pass
+        except TraitError:
+            # They entered something invalid, pretend it didn't happen
+            value = self.value
+            self.control.text.setText(str(value))
```

Comment:
Without this my test was failing because we would set text to 0.0, hit enter, error message pops up, we click ok on the error dialog.  The slider value would be unchanged (eg still show as 0.5 because the error occurred in the `self.value = value` line above so we never got to the `self.control.slider.setValue` line.  However, the textbox would still contain what the user had been typing just before hitting enter.  In other words it would still show 0.0.  Then, when the main ui was closed (we exit the `create_ui` context manager), it was as if we were trying to enter that value in the text field again.  This led to _another_ error dialog causing the test to hang.

I think reseting the text box to the previously set _valid_ value before you hit enter makes sense from a UX perspective.  Perhaps a reviewer will disagree though?  If so, I can try to modify the code / test to work with that UX.  If it doesn't get reset then it is expected that the user will modify the value again to an acceptable value.

Reply:
Where resetting the value can be problematic is when you are transitioning through an invalid value to enter a valid value, eg. if the minimum was `0.05` and you started typing `0.0` on the way to that, having a reset makes it hard to hit the desired value.

However if this is only ever triggered by ending editing, resetting is OK, as by definition once editing is done the user can't be transitioning to another value.